router_types.c

```diff
@@ -44,6 +49,53 @@ static const struct router_type *router_types[] = {
 	NULL,
 };
 
+static int read_mac(uint8_t mac[ETH_ALEN], const char *macstr)
+{
+	int ret;
+
+	ret = sscanf(macstr, "%02hhX:%02hhX:%02hhX:%02hhX:%02hhX:%02hhX",
+		     &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+	if (ret != 6)
+		ret = sscanf(macstr, "%02hhX-%02hhX-%02hhX-%02hhX-%02hhX-%02hhX",
+			     &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+
+	return (ret == 6);
+}
+
+int mac_whitelist_add(const char *macstr)
+{
+	uint8_t mac[ETH_ALEN];
+	struct mac_whitelist_entry *new_entry = malloc(sizeof(*new_entry));
```

Comment:
no, this is not what I meant. The 

    new_entry = malloc(sizeof(*new_entry));

Should be right before

    if (!new_entry) {

and the

    struct mac_whitelist_entry *new_entry;

should be right here.